src/__swaps__/screens/Swap/components/ReviewPanel.tsx

```diff
@@ -20,135 +19,65 @@ import { chainNameFromChainIdWorklet } from '@/__swaps__/utils/chains';
-import { useAccountSettings } from '@/hooks';
-import { FormattedExternalAsset } from '@/resources/assets/externalAssetsQuery';
-import { supportedNativeCurrencies } from '@/references';
 
 const SLIPPAGE_STEP = 0.5;
 
 const RainbowFee = () => {
   const { isDarkMode } = useColorMode();
-  const { SwapInputController } = useSwapContext();
-  const { nativeCurrency: currentCurrency } = useAccountSettings();
 
   const rainbowFee = useSharedValue(i18n.t(i18n.l.swap.unknown));
 
-  const [nativeChainId, setNativeChainId] = useState(SwapInputController.assetToSell.value?.chainId ?? ChainId.mainnet);
-  const nativeAsset = useNativeAssetForNetwork(ethereumUtils.getNetworkFromChainId(nativeChainId));
-
-  const updateRainbowFee = ({
-    fee,
-    nativeAsset,
-    currentCurrency,
-  }: {
-    fee: string | number;
-    nativeAsset: FormattedExternalAsset | null | undefined;
-    currentCurrency: keyof typeof supportedNativeCurrencies;
-  }) => {
-    const updateFee = (value: string) => {
-      'worklet';
-      rainbowFee.value = value;
-    };
-
-    const { display } = convertRawAmountToNativeDisplay(
-      fee,
-      nativeAsset?.decimals || 18,
-      nativeAsset?.price?.value || '0',
-      currentCurrency
-    );
-
-    runOnUI(updateFee)(display);
-  };
-
-  useAnimatedReaction(
-    () => SwapInputController.assetToSell.value?.chainId ?? ChainId.mainnet,
-    (current, previous) => {
-      if (!previous || previous !== current) {
-        runOnJS(setNativeChainId)(current);
-      }
-    }
-  );
-
-  useAnimatedReaction(
-    () => ({
-      fee: SwapInputController.fee.value,
-      nativeAsset,
-      currentCurrency,
-    }),
-    (current, previous) => {
-      if (
-        !previous ||
-        previous.nativeAsset !== current.nativeAsset ||
-        previous.currentCurrency !== current.currentCurrency ||
-        previous.fee !== current.fee
-      ) {
-        runOnJS(updateRainbowFee)({
-          fee: current.fee,
-          nativeAsset: current.nativeAsset,
-          currentCurrency: current.currentCurrency,
-        });
-      }
-    }
-  );
-
   return <AnimatedText align="right" color={isDarkMode ? 'labelSecondary' : 'label'} size="15pt" weight="heavy" text={rainbowFee} />;
 };
 
 export function ReviewPanel() {
   const { isDarkMode } = useColorMode();
-  const { reviewProgress, SwapInputController } = useSwapContext();
+  const { reviewProgress, SwapInputController, internalSelectedInputAsset, internalSelectedOutputAsset } = useSwapContext();
 
   const unknown = i18n.t(i18n.l.swap.unknown);
 
   const chainName = useDerivedValue(() =>
-    SwapInputController.outputChainId.value === ChainId.mainnet
+    internalSelectedOutputAsset.value?.chainId === ChainId.mainnet
       ? 'ethereum'
-      : chainNameFromChainIdWorklet(SwapInputController.outputChainId.value)
+      : chainNameFromChainIdWorklet(internalSelectedOutputAsset.value?.chainId ?? ChainId.mainnet)
```

Comment:
if chainId falls back to mainnet, should we set chainName to 'ethereum' as we do on line 41 instead of calling chainNameFromChainIdWorklet?

Reply:
yes tbh review panel is going to need hooked up in a follow-up PR. nothing done to this except gutting the old asset stuff out.

Reply:
I can add it though